### PR TITLE
Remove some usage of MatrixCallback

### PR DESCRIPTION
**Pull request by bmarty**
I have reverted the change on the crypto part as we are not sure about the side effect.

The rest seems OK, after a quick smoke test.

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt [hunk 1]
```diff
@@ -1320,16 +1320,15 @@ internal class DefaultCryptoService @Inject constructor(
...
         cryptoCoroutineScope.launch(coroutineDispatchers.crypto) {
             Timber.tag(loggerTag.value).d("prepareToEncrypt() roomId:$roomId Check room members up to date")
             // Ensure to load all room members
             try {
                 loadRoomMembersTask.execute(LoadRoomMembersTask.Params(roomId))
             } catch (failure: Throwable) {
                 Timber.tag(loggerTag.value).e("prepareToEncrypt() : Failed to load room members")
-                callback.onFailure(failure)
-                return@launch
+                throw failure
```
Comment (ouchadam) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt:1331]: If I've understood correctly, throwing here shouldn't have any behaviour changes as it's the same as `awaitCallback`?

**Review by ouchadam — APPROVED** [review]
assuming there aren't more changes on the way!

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt:1331]: I am not sure :/
I will do more test, or I should say I will do test 🙈

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt:1331]: The app crashes :/

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt]: @ouchadam I have reverted this change as we (V and G) are not sure about the impact on the crypto code here.

Comment (ouchadam) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt]: thanks for checking! :+1:

File: vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt [hunk 2]
```diff
@@ -740,14 +740,22 @@ class TimelineViewModel @AssistedInject constructor(
     }
 
     private fun handleUndoReact(action: RoomDetailAction.UndoReaction) {
-        room.undoReaction(action.targetEventId, action.reaction)
+        viewModelScope.launch {
+            tryOrNull {
+                room.undoReaction(action.targetEventId, action.reaction)
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt:745]: nit: could be handy to log the error case

**Review by ouchadam — APPROVED** [review]
no blockers from me, looks good! :+1:

**Pull request merged**